### PR TITLE
fix: duplicate command completion

### LINKED ISSUE #411: [Bug]: Console suggestions are repeated once per alias
### Tempest Version

1.0-alpha1

### PHP Version

8.3

### Operating System

MacOS

### Description

```
[ ] discovery:cache
[ ] discovery:cache
[x] discovery:status
[ ] discovery:status
[ ] discovery:clear
[ ] discovery:clear
```

### Steps to Reproduce

/

### PR DESCRIPTION
**Pull request by Treggats**
fixes #411

Worked on this with @Bapawe 

This PR removes duplicate completions for commands

**Steps to reproduce**
Like mentioned in the issue run
```shell
php tempest di
```
Expected output
```shell
~/P/tempest-framework ❯❯❯ php tempest di
 Command di not found 
 Did you mean to run one of these? 
[x] fail
[ ] discovery:status
[ ] discovery:clear
[ ] tail
Press enter to confirm, ctrl+c to cancel
```

### CODE REVIEW

File: src/Tempest/Console/src/Testing/ConsoleTester.php [hunk 1]
```diff
@@ -160,6 +160,19 @@ public function printFormatted(): self
         return $this;
     }
 
+    public function getContent(?callable $callback = null): array
```
Comment (Treggats) [inline src/Tempest/Console/src/Testing/ConsoleTester.php:163]: helper to get the output for the console

File: src/Tempest/Console/src/Middleware/ResolveOrRescueMiddleware.php [hunk 2]
```diff
@@ -65,6 +65,10 @@ private function getSimilarCommands(string $name): array
         $similarCommands = [];
 
         foreach ($this->consoleConfig->commands as $consoleCommand) {
+            if (in_array($consoleCommand->getName(), $similarCommands, strict: true)) {
```
Comment (Treggats) [inline src/Tempest/Console/src/Middleware/ResolveOrRescueMiddleware.php:68]: if the command name is already is in the list, then don't add it again

File: tests/Integration/Console/Middleware/ResolveOrRescueMiddlewareTest.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    #[Test]
+    public function it_can_find_multiple_similar_commands(): void
+    {
+        $this->console
+            ->call('discovery')
+            ->assertSee('Did you mean to run one of these?  [discovery:status/discovery:clear]');
+    }
+
+    #[Test]
+    public function it_does_not_duplicate_completed_commands(): void
```
Comment (Treggats) [inline tests/Integration/Console/Middleware/ResolveOrRescueMiddlewareTest.php:36]: similar to the previous test, but this test more the inner workings

File: tests/Integration/Console/Middleware/ResolveOrRescueMiddlewareTest.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $formatOutput = static fn (string $buffer) => str($buffer)
+            ->trim()
+            ->remove(['[',']'])
+            ->explode('/')
+            ->all();
+
+        $output = $this->console
+            ->call('discovery')
+            ->getContent(fn (array $buffer) => $formatOutput(array_pop($buffer)));
```
Comment (brendt) [inline tests/Integration/Console/Middleware/ResolveOrRescueMiddlewareTest.php:38]: Maybe better to call this `getOutput`?

**Review by brendt — CHANGES_REQUESTED** [review]

**Comment by brendt** [comment]
Looks good! Take a look at my one comment and rebase on main, after that this is good to merge

File: tests/Integration/Console/Middleware/ResolveOrRescueMiddlewareTest.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $formatOutput = static fn (string $buffer) => str($buffer)
+            ->trim()
+            ->remove(['[',']'])
+            ->explode('/')
+            ->all();
+
+        $output = $this->console
+            ->call('discovery')
+            ->getBuffer(fn (array $buffer) => $formatOutput(array_pop($buffer)));
```
Comment (Treggats) [inline tests/Integration/Console/Middleware/ResolveOrRescueMiddlewareTest.php:38]: I think that `getBuffer()` aptly describes the output it returns

**Review by brendt — APPROVED** [review]

**Comment by brendt** [comment]
Thanks!

**Pull request merged**